core/src/test/java/com/graphhopper/routing/util/BikeFlagEncoderTest.java

```diff
@@ -567,18 +567,20 @@ public void testSacScale() {
 
     @Test
     public void testCalcPriority() {
+        // we want to test the priority assuming the way is accessible, even when it is not
         ReaderWay osmWay = new ReaderWay(1);
+        osmWay.setTag("highway", "tertiary");
         ReaderRelation osmRel = new ReaderRelation(1);
         osmRel.setTag("route", "bicycle");
         osmRel.setTag("network", "icn");
         IntsRef relFlags = encodingManager.handleRelationTags(osmRel, encodingManager.createRelationFlags());
-        IntsRef flags = encodingManager.handleWayTags(osmWay, accessMap, relFlags);
+        IntsRef flags = encodingManager.handleWayTags(osmWay, relFlags);
         assertEquals(PriorityCode.getValue(BEST.getValue()), priorityEnc.getDecimal(false, flags), .1);
 
         // important: UNCHANGED should not get 0 priority!
         osmWay = new ReaderWay(1);
-        osmWay.setTag("highway", "somethingelse");
-        flags = encodingManager.handleWayTags(osmWay, accessMap, encodingManager.createRelationFlags());
+        osmWay.setTag("highway", "track");
+        flags = encodingManager.handleWayTags(osmWay, encodingManager.createRelationFlags());
         assertEquals(PriorityCode.getValue(UNCHANGED.getValue()), priorityEnc.getDecimal(false, flags), .1);
```

Comment:
@karussell do you remember what this comment 

```
// important: UNCHANGED should not get 0 priority!
```

is supposed to say? The issue with this test is that previously we called `AbstractFlagEncoder#getAccess` before `handleWayTags` and passed the `AcceptWay` into `handleWayTags`. This allowed us in this test to pass `Access=WAY` and test the priority even for ways that aren't even accessible. This is why the test used to work even for `highway=somethingelse`. I now changed it to `track` because then we also get `priority==UNCHANGED`. But I don't understand why the comment says the priority should not be 0, but then we do not even test whether the priority is not 0...

Reply:
I think I meant that the default value is 0. And the default meaning should be "unchanged priority" but we do not have an option for this (anymore) and we need to set this UNCHANGED explicitly. But not sure.

Should we ensure that `priority` is UNCHANGED for unknown highway values (?)

Reply:
> I think I meant that the default value is 0.

Ok, this makes sense. It *does* get the default value 0 as long as the access is false though.

> Should we ensure that priority is UNCHANGED for unknown highway values (?)

Probably, but maybe this is not needed yet (at least for this PR), because for unknown highway values the access will be false anyway. The test only works in master because we pass `Access.WAY` even though in reality it would be `CAN_SKIP`.